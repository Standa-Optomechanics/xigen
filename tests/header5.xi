protocol "v1"
defaults with crc, answer, public

command "foobar" universal "foo" (40)
fields:
	calb float position
  normal int32s position
	normal int16s uposition
	int16s test
	char strfoo[6]
	int16s arr[8]
	char strbar[4]

command "bazqux" universal "qux" (30)
with inline
fields:
	calb float position
  normal int32s position
	normal int16s uposition
	int16s test
	int16s arr[8]

command "repeated" universal "rpt" (24)
with inline
fields:
	calb float position
  normal int32s position
	normal int16s uposition
	int16s test
	calb float speed
  normal int32s speed
	normal int16s uspeed
	float another

command "foobar2" universal "foo" (28)
with inline
fields:
	calb float position
  normal int32s position
	int16s test
	int16s arr[8]

command "foobar3" universal "foo" (36)
with inline
fields:
	calb float position
  normal int32s position
	calb float speed
  normal int32s speed
  normal int32s microspeed
	int16s test
	int16s arr[8]

command "arr1" universal "arr" (72)
with inline
fields:
	calb float position[8]
  normal int32s position[8]
  normal int32s microposition[8]
	int16s test

command "arr2" universal "arr" (72)
fields:
	calb float position[8]
  normal int32s position[8]
  normal int32s microposition[8]
	int16s test

command "arr3" universal "arr" (48)
fields:
	calb float position[8]
  normal int32s position[8]
  normal int8s microposition[8]
	int16s test

command "byted" universal "byt" (64)
fields:
	calb float position[8]
  normal int32s position[8]
  normal int8s microposition[8]
  byte key[16]
	int16s test

