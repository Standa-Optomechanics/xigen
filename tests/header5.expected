
/** @cond DO_NOT_WANT */
#define LIBXIMC_PROTOCOL_VERSION 1
/** @endcond */


/*
 ------------------------------------------
   BEGIN OF GENERATED struct declarations
 ------------------------------------------
*/

	typedef struct
	{
		int position;
		int uposition;
		int test;
		char strfoo[7];
		int arr[8];
		char strbar[5];
	} foobar_t;

	typedef struct
	{
		float position;
		int test;
		char strfoo[7];
		int arr[8];
		char strbar[5];
	} foobar_calb_t;

	typedef struct
	{
		int position[8];
		int microposition[8];
		int test;
	} arr2_t;

	typedef struct
	{
		float position[8];
		int test;
	} arr2_calb_t;

	typedef struct
	{
		int position[8];
		int microposition[8];
		int test;
	} arr3_t;

	typedef struct
	{
		float position[8];
		int test;
	} arr3_calb_t;

	typedef struct
	{
		int position[8];
		int microposition[8];
		uint8_t key[16];
		int test;
	} byted_t;

	typedef struct
	{
		float position[8];
		uint8_t key[16];
		int test;
	} byted_calb_t;

/*
 --------------------------------------------
   BEGIN OF GENERATED function declarations
 --------------------------------------------
*/
	result_t XIMC_API set_foobar (device_t id, const foobar_t* foobar);

	result_t XIMC_API set_foobar_calb (device_t id, const foobar_calb_t* foobar_calb, const calibration_t* calibration);

	result_t XIMC_API get_foobar (device_t id, foobar_t* foobar);

	result_t XIMC_API get_foobar_calb (device_t id, foobar_calb_t* foobar_calb, const calibration_t* calibration);

	result_t XIMC_API set_bazqux (device_t id, int position, int uposition, int test, const int* arr);

	result_t XIMC_API set_bazqux_calb (device_t id, float position, int test, const int* arr, const calibration_t* calibration);

	result_t XIMC_API get_bazqux (device_t id, int* position, int* uposition, int* test, int* arr);

	result_t XIMC_API get_bazqux_calb (device_t id, float* position, int* test, int* arr, const calibration_t* calibration);

	result_t XIMC_API set_repeated (device_t id, int position, int uposition, int test, int speed, int uspeed, float another);

	result_t XIMC_API set_repeated_calb (device_t id, float position, int test, float speed, float another, const calibration_t* calibration);

	result_t XIMC_API get_repeated (device_t id, int* position, int* uposition, int* test, int* speed, int* uspeed, float* another);

	result_t XIMC_API get_repeated_calb (device_t id, float* position, int* test, float* speed, float* another, const calibration_t* calibration);

	result_t XIMC_API set_foobar2 (device_t id, int position, int test, const int* arr);

	result_t XIMC_API set_foobar2_calb (device_t id, float position, int test, const int* arr, const calibration_t* calibration);

	result_t XIMC_API get_foobar2 (device_t id, int* position, int* test, int* arr);

	result_t XIMC_API get_foobar2_calb (device_t id, float* position, int* test, int* arr, const calibration_t* calibration);

	result_t XIMC_API set_foobar3 (device_t id, int position, int speed, int microspeed, int test, const int* arr);

	result_t XIMC_API set_foobar3_calb (device_t id, float position, float speed, int test, const int* arr, const calibration_t* calibration);

	result_t XIMC_API get_foobar3 (device_t id, int* position, int* speed, int* microspeed, int* test, int* arr);

	result_t XIMC_API get_foobar3_calb (device_t id, float* position, float* speed, int* test, int* arr, const calibration_t* calibration);

	result_t XIMC_API set_arr1 (device_t id, const int* position, const int* microposition, int test);

	result_t XIMC_API set_arr1_calb (device_t id, const float* position, int test, const calibration_t* calibration);

	result_t XIMC_API get_arr1 (device_t id, int* position, int* microposition, int* test);

	result_t XIMC_API get_arr1_calb (device_t id, float* position, int* test, const calibration_t* calibration);

	result_t XIMC_API set_arr2 (device_t id, const arr2_t* arr2);

	result_t XIMC_API set_arr2_calb (device_t id, const arr2_calb_t* arr2_calb, const calibration_t* calibration);

	result_t XIMC_API get_arr2 (device_t id, arr2_t* arr2);

	result_t XIMC_API get_arr2_calb (device_t id, arr2_calb_t* arr2_calb, const calibration_t* calibration);

	result_t XIMC_API set_arr3 (device_t id, const arr3_t* arr3);

	result_t XIMC_API set_arr3_calb (device_t id, const arr3_calb_t* arr3_calb, const calibration_t* calibration);

	result_t XIMC_API get_arr3 (device_t id, arr3_t* arr3);

	result_t XIMC_API get_arr3_calb (device_t id, arr3_calb_t* arr3_calb, const calibration_t* calibration);

	result_t XIMC_API set_byted (device_t id, const byted_t* byted);

	result_t XIMC_API set_byted_calb (device_t id, const byted_calb_t* byted_calb, const calibration_t* calibration);

	result_t XIMC_API get_byted (device_t id, byted_t* byted);

	result_t XIMC_API get_byted_calb (device_t id, byted_calb_t* byted_calb, const calibration_t* calibration);


/*
 -------------------------
   END OF GENERATED CODE
 -------------------------
*/
